src/renderer/vars/setup-build-version.injectable.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { runInAction } from "mobx";
+import { requestFromChannelInjectionToken } from "../../common/utils/channel/request-from-channel-injection-token";
+import { buildVersionChannel, buildVersionInjectionToken } from "../../common/vars/build-semantic-version.injectable";
+import { beforeFrameStartsInjectionToken } from "../before-frame-starts/before-frame-starts-injection-token";
+
+const setupBuildVersionInjectable = getInjectable({
+  id: "setup-build-version",
+  instantiate: (di) => {
+    const requestFromChannel = di.inject(requestFromChannelInjectionToken);
+
+    return {
+      run: async () => {
+        const buildVersion = await requestFromChannel(buildVersionChannel);
+
+        runInAction(() => {
+          di.register(getInjectable({
```

Comment:
Using the di-container to store a sync state from async source is not the responsibility of di.

Instead, we could create an abstraction for this particular responsibility, eg. `syncAsync`, or `settableGettable` or `setuppableState`. This would also remove duplication from places such as `lens-proxy-port.injectable.ts` and `app-paths-state.injectable.ts`.

Reply:
Oh okay, I thought you would have wanted to put all of these sorts of states into the DI. Will change.